packages/hint-compat-api/src/helpers/compat-html.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * @fileoverview Helper that contains all the logic related with HTML compat api, to use in different modules.
+ */
+
+import { MDNTreeFilteredByBrowsers, TestFeatureFunction, FeatureInfo } from '../types';
+import { HintContext } from 'hint/dist/src/lib/hint-context';
+import { ElementFound, IAsyncHTMLElement, ProblemLocation, AsyncHTMLAttribute, IAsyncNamedNodeMap } from 'hint/dist/src/lib/types';
+import { CompatStatement } from '../types-mdn.temp';
+import { HTMLParse, HTMLEvents } from '@hint/parser-html/dist/src/types';
+import { CompatBase } from './compat-base';
+
+export class CompatHTML extends CompatBase<HTMLEvents, HTMLParse> {
+    public constructor(hintContext: HintContext<HTMLEvents>, MDNData: MDNTreeFilteredByBrowsers, testFunction: TestFeatureFunction) {
+        super(hintContext, MDNData, testFunction);
+
+        hintContext.on('parse::end::html', this.onParse.bind(this));
+    }
+
+    public async searchFeatures(parser: HTMLParse): Promise<void> {
+        await this.walk(async (elementFound: ElementFound) => {
+            const { element } = elementFound;
+            const location = element.getLocation();
+            /*
+             * WORKAROUND: Element listener is being fired twice. The second
```

Comment:
I'm just curious. Why is the listener fired twice?

Reply:
To be honest, I didn’t spend too much time investigating  the issue. I will take a look and let you know

Reply:
ok!

Reply:
So after digging into the issue I found the `element::$tagName` event is fired from two different places:
-  `packages\connector-jsdom\src\connector.ts`. It doesn't include the location.
- `packages\parser-html\src\parser.ts`. It includes the location.

I'm not sure if this is expected or not, but it sounds weird to me.

Reply:
> I'm not sure if this is expected or not, but it sounds weird to me.

That sounds like you are using both in your configuration file, the connector `jsdom` and the html parser.

IIRC, the html parser is mean to be use just with the `local` connector, @antross am I right?

Reply:
We should do something to avoid this:
- Make the configuration fail (not a huge fan)
- Make the parser smarter and disable itself if it receives one of the events it's supposed to send?
- Something else?

Reply:
FYI: If we go for the second option (it seems smarter), we should take into account the event sent from the connector is not populating the location

Reply:
In any case, I guess this issue is not related with the compat-api-html implementation. Maybe we should deal with this in a different branch.

Reply:
> Maybe we should deal with this in a different branch.

Agreed. Glad we found it, though. For this PR, you should be able to fix your local configuration and remove the workaround. I've filed #1593 to track fixing `parser-html` to prevent this.

Reply:
> FYI: If we go for the second option (it seems smarter), we should take into account the event sent from the connector is not populating the location

Also agreed. The `report` API will "infer" a location if passed an `element` (one of the optional parameters). In fact behind the scenes this calls `element.getLocation` first before trying to infer the location. 

Some of this is for historical reasons and it may be better at some point to extract this inference into a helper and update connectors to use it as a fallback in `element.getLocation` when no parser-provided location is known. However, this is how things work today.

Reply:
In order to skip the `jsdom` connector I had to modify the `connectors.ts` file removing the `jsdom`option. But this is not a custom configuration, this is the default configuration in the repo.

```typescript
packages\utils-tests-helpers\src\connectors.ts

export const ids = [
    'chrome',
    'jsdom'
];
```

After this update I can confirm the workaround is not needed anymore.

@antross About your last message.

> You should be able to fix your local configuration and remove the workaround.

I'm not sure if there is a way to fix my local configuration without updating any file in the repo. Am I missing something?

Reply:
> I'm not sure if there is a way to fix my local configuration without updating any file in the repo.

I saw the issue and sorry I missed it in my earlier reviews. We need to exclude specifying `parsers: ['html']` in the configuration for the test cases to fix this (it's not needed). I added a review suggestion on one of the tests so you can see exactly what I mean.

Reply:
>In order to skip the jsdom connector I had to modify the connectors.ts file removing the jsdomoption. But this is not a custom configuration, this is the default configuration in the repo.

> I saw the issue and sorry I missed it in my earlier reviews. We need to exclude specifying parsers: ['html'] in the configuration for the test cases to fix this (it's not needed). I added a review suggestion on one of the tests so you can see exactly what I mean.

Maybe this is not the case, but just in case @borgitas21 doesn't know it. You can configure your tests to skip one connector or another, add parsers, etc. for example, here we are ignoring the `chrome` connector: https://github.com/webhintio/hint/blob/master/packages/hint-http-compression/tests/tests-http.ts#L23

Reply:
Cool 😀. Thank you guys!